crypto/src/hash.rs

```diff
@@ -224,6 +222,8 @@ impl<T> Clone for HashOf<T> {
 impl<T> Copy for HashOf<T> {}
 
 impl<T> PartialEq for HashOf<T> {
+    // NOTE: False positive
+    #[allow(clippy::unconditional_recursion)]
```

Comment:
Yeeah, this one is definitely because of the night (and fixed already: https://github.com/rust-lang/rust-clippy/issues/12133)

Reply:
that's why I said if we could somehow upgrade to a nightly version that will become next stable. But I'm not sure that Rust's release cycle works that way

Reply:
I don't know much about the versioning of clippy, but rustc doesn't work this way. They branch it as a beta, backport bugfixes to it and then promote it to stable. 

Clippy seems to have a somewhat similar process: https://doc.rust-lang.org/nightly/clippy/development/infrastructure/release.html